src/strategy/strategy.ts

```diff
@@ -166,6 +166,9 @@ operators.set(Operator.DATE_BEFORE, DateOperator);
 operators.set(Operator.SEMVER_EQ, SemverOperator);
 operators.set(Operator.SEMVER_GT, SemverOperator);
 operators.set(Operator.SEMVER_LT, SemverOperator);
+
+export type StrategyResult = { enabled: true, variant?: Variant | null } | { enabled: false };
```

Comment:
Why is variant both undefineable and null? Do we treat them differently? That seems subtle and like something that might cause issues down the line. Why not just undefined?

```suggestion
export type StrategyResult = { enabled: true, variant?: Variant } | { enabled: false };
```

Reply:
One of the methods that we call (selecting variant from a list) returns null so we have to handle this

Reply:
Ah. In that case, could we just make it `Variant | null`? Because we allow both it strikes me as significant, so  I would expect `undefined` and `null` to mean different things. If they don't, I would prefer collapsing it into a single value.

Your choice in the end, but the more we can lock it down and be strict about what we return, the easier it'll be to work with later.

Reply:
After further investigation we can remove null